libs/libloggers/loggers/OwnPatternFormatter.cpp

```diff
@@ -4,15 +4,72 @@
 #include <optional>
 #include <IO/WriteBufferFromString.h>
 #include <IO/WriteHelpers.h>
+#include <Common/HashTable/Hash.h>
 #include <Interpreters/InternalTextLogsQueue.h>
 #include <sys/time.h>
 #include <Common/CurrentThread.h>
 #include <common/getThreadId.h>
 #include "Loggers.h"
 
 
-OwnPatternFormatter::OwnPatternFormatter(const Loggers * loggers_, OwnPatternFormatter::Options options_)
-    : Poco::PatternFormatter(""), loggers(loggers_), options(options_)
+static const char * setColor(UInt64 num)
+{
+    /// ANSI escape sequences to set foreground font color in terminal.
+
+    static constexpr auto num_colors = 14;
+    static const char * colors[num_colors] =
+    {
+        /// Black on black is meaningless
+        "\033[0;31m",
+        "\033[0;32m",
+        "\033[0;33m",
+        /// Low intense blue on black is too dark.
+        "\033[0;35m",
+        "\033[0;36m",
+        "\033[0;37m",
+        "\033[1;30m",
```

Comment:
And this one also will looks like there is no text (like trace level priority)

Reply:
> force black background for all colors.

If you are Ok with this variant of solution, could you please experiment and send a PR?

Reply:
I don't "like" one-linear PR's, but okay will take a look a little bit later (if you will not fix it yet)

Reply:
Forcing everything into black BG leads to unpleasant results:

![2020-02-06 18 32 52](https://user-images.githubusercontent.com/9271526/73951890-50f82c00-490f-11ea-92a9-01d6f76b518f.jpg)
![2020-02-06 18 32 45](https://user-images.githubusercontent.com/9271526/73951894-52c1ef80-490f-11ea-8f32-b0aced2e7c6e.jpg)

Reply:
We better not use pure black/white as FG color and rely on what every color scheme has to offer